app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/validateemail/ValidateUserEmailViewModel.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.screen.newtransfer.validateemail
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.multiplatform_swisstransfer.managers.UploadManager
+import com.infomaniak.multiplatform_swisstransfer.network.exceptions.EmailValidationException
+import com.infomaniak.multiplatform_swisstransfer.network.models.upload.request.VerifyEmailCodeBody
+import com.infomaniak.multiplatform_swisstransfer.network.repositories.UploadRepository
+import com.infomaniak.swisstransfer.di.IoDispatcher
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ValidateUserEmailViewModel @Inject constructor(
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+    private val uploadManager: UploadManager
+) : ViewModel() {
+
+    private val uploadRepository: UploadRepository by lazy { UploadRepository() }
+
+    private val _uiState = MutableStateFlow(ValidateEmailUiState.Default)
+    val uiState = _uiState.asStateFlow()
+
+    fun validateEmailWithOtpCode(email: String, otpCode: String, onSuccess: () -> Unit, onUnknownError: () -> Unit) {
+        viewModelScope.launch(ioDispatcher) {
+            _uiState.value = ValidateEmailUiState.Loading
+
+            runCatching {
+                uploadRepository.verifyEmailCode(VerifyEmailCodeBody(otpCode, email))
```

Comment:
Since you're using the `uploadManager` here, maybe we can add a method in  `uploadManager` to call the `verifyEmailCode` ? `uploadManager` is used as a common access point for the repository and the controller of uploads so it might be cleaner ?

And if we're a doing that on KMP, we can directly save the token in DB, instead of doing that line 57. Let me know if it'll cause any issue with iOS, as we discussed yesterday as it might interfere with what they did on the native iOS side of things.

Also, we're using the manager to do `uploadManager.resendEmailCode(email)` so it might be good to do the same thing for everything else.